src/main/java/org/jenkinsci/plugins/github_branch_source/RepositoryUriResolver.java

```diff
@@ -39,14 +35,13 @@
     public abstract String getRepositoryUri(String apiUri, String owner, String repository);
 
     public static String hostnameFromApiUri(String apiUri) {
-        apiUri = Util.fixEmptyAndTrim(apiUri);
-        try {
-            URL endpoint = new URL(apiUri);
-            if (endpoint.getPath().startsWith("/api/v3")) {
+        if (apiUri != null) {
+            try {
+                URL endpoint = new URL(apiUri);
                 return endpoint.getHost();
+            } catch (MalformedURLException e) {
+                // ignore
```

Comment:
This happens when the logic (backend) defines a behavior and the UI other different :dizzy_face: 
The idea is including input data validation to avoid this kind of checks here.